p2panda-auth/src/group/test_utils/group_store.rs

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::rc::Rc;
+
+use thiserror::Error;
+
+use crate::traits::{GroupStore, IdentityHandle};
+
+use super::TestGroupState;
+
+#[derive(Debug, Error)]
+pub enum GroupStoreError {}
+
+#[derive(Clone, Debug)]
+pub struct TestGroupStore<ID>(Rc<RefCell<HashMap<ID, TestGroupState>>>)
+where
+    ID: IdentityHandle;
+
+impl<ID> Default for TestGroupStore<ID>
+where
+    ID: IdentityHandle,
+{
+    fn default() -> Self {
+        Self(Rc::new(RefCell::new(HashMap::new())))
+    }
+}
+
+impl GroupStore<char> for TestGroupStore<char> {
+    type Group = TestGroupState;
```

Comment:
Not sure if this needs to be generic. This will almost always be a `GroupState` or not?